fix computing nt from GlobalDimensions

nils-asmussen opened the pull request:
Fixes a regression in Hadrons/DilutedNoise.hpp and Test_qed.cc that leads to an incorrect value for the number of time slices.

aportelli commented:
Could you give a bit more context on how this occurred?
Anyway that looks a bit to verbose to me, just using `.back()` would do. Checking the dimension array size is an overkill, there is no way a grid would have been created with an empty dimension array (or the programme would have crashed long before).

nils-asmussen commented:
In commit 61d017d0a5ea12e3798b759af7d9b4c5acdbb13c, in the file Hadrons/DilutedNoise.hpp in the function TimeDilutedSpinColorDiagonalNoise, the line
```    nt_ = this->getGrid()->GlobalDimensions().back();```
was changed to
```    nt_ = this->getGrid()->GlobalDimensions().size();```

I guess that the reason for the change was, that the type AcceleratorVector of GlobalDimensions() does not implement a back() member function.

A similar change occurred in tests/core/Test_qed.cc, that I also fixed.

I have removed the superfluous assertion.

aportelli commented:
Obsolete
